pkg/cluster/cache.go

```diff
@@ -62,12 +67,36 @@ func (c *kubeFedClusterClients) getFedCluster(name string) (*FedCluster, bool) {
 	return cluster, ok
 }
 
-func (c *kubeFedClusterClients) getFedClustersByType(clusterType Type) []*FedCluster {
+// Condition an expected cluster condition
+type Condition func(cluster *FedCluster) bool
+
+// Ready checks that the cluster is in a 'Ready' status condition
+var Ready Condition = func(cluster *FedCluster) bool {
+	for _, condition := range cluster.ClusterStatus.Conditions {
+		if condition.Type == common.ClusterReady && condition.Status == apiv1.ConditionTrue {
+			return true
+		}
+	}
+	return false
```

Comment:
you can replace the whole content of the function by:
```
return util.IsClusterReady(cluster.ClusterStatus)
```

Reply:
ok, will do

Reply:
done in 1222e2d